### PR TITLE
add `--unprocessed-images-inspect` flag

### PR DESCRIPTION
**Pull request by pivotaljohn**
Sketch for conversation: https://kubernetes.slack.com/archives/CH8KCCKA5/p1630636823106500

### CODE REVIEW

File: pkg/kbld/cmd/resolve.go [hunk 1]
```diff
@@ -78,7 +80,20 @@ func (o *ResolveOptions) Run() error {
 
 	imgFactory := ctlimg.NewFactory(conf, registry, logger)
 
-	resolvedImages, err := o.resolveImages(nonConfigRs, conf, imgFactory)
+	imageURLs, err := o.collectImageReferences(nonConfigRs, conf)
+	if err != nil {
+		return err
+	}
+	if o.UnprocessedImageInspect {
```
Comment (cppforlife) [inline pkg/kbld/cmd/resolve.go]: nit: add newline before

File: pkg/kbld/cmd/unprocessed_image_urls.go [hunk 2]
```diff
@@ -33,3 +35,7 @@ func (i *UnprocessedImageURLs) All() []UnprocessedImageURL {
 	})
 	return result
 }
+
+func (i *UnprocessedImageURLs) Bytes() ([]byte, error) {
```
Comment (cppforlife) [inline pkg/kbld/cmd/unprocessed_image_urls.go:39]: dont think we want to use `url: ...` in its output. prolly better to go with `image: ...` (locking down serialized format independently from UnprocessedImageURL struct might be worth it?)

**Comment by cppforlife** [comment]
any thoughts on naming and output for processed images?

File: pkg/kbld/cmd/unprocessed_image_urls.go (hunk 2 above)
Comment (pivotaljohn) [reply pkg/kbld/cmd/unprocessed_image_urls.go:39]: I think, for now, placing the json struct tags on the internal data structure is _sufficient_ for now. Might be something to revisit as other requests (like #95) are addressed.

**Comment by pivotaljohn** [comment]
> any thoughts on naming and output for processed images?

Updated per suggestions. E2E tests are running (takes ~30min); these changes should have no affect on them.

File: pkg/kbld/cmd/unprocessed_image_urls.go [hunk 3]
```diff
@@ -5,14 +5,16 @@ package cmd
...
 )
 
 type UnprocessedImageURL struct {
-	URL string
+	URL string `json:"image"`
 }
 
 type UnprocessedImageURLs struct {
-	urls map[UnprocessedImageURL]struct{}
+	urls map[UnprocessedImageURL]struct{} `json:"unresolved"`
```
Comment (cppforlife) [inline pkg/kbld/cmd/unprocessed_image_urls.go:17]: hmm, ytt doesnt have map key as top level when doing --files-inspect/--data-values-inspect. but its also probably a good idea to have it... should we rename this to `images`?

Comment (pivotaljohn) [reply pkg/kbld/cmd/unprocessed_image_urls.go:17]: Chose "unresolved" to make clear the status. But, I can see more useful if the data structure were consistent (e.g. if/when another inspect is inserted later in the processing).

Whether a top-level _should_ be present: is it possible we'd like to output a mixture of objects as "current state" in this or subsequent inspects? If yes, then the top-level key is needed; if not, then its presence is somewhat dubious.

**Pull request merged**